File: mobile/lib/features/patients/models/update_patient_medical_record_request.dart
class UpdatePatientMedicalRecordRequestModel {
  UpdatePatientMedicalRecordRequestModel({
    required this.recordTypeId,
    required this.recordName,
    required this.reportDate,
    this.comments,
  });

  final int recordTypeId;
  final String recordName;
  final DateTime reportDate;
  final String? comments;

  Map<String, dynamic> toJson() {
    return {
      'recordTypeId': recordTypeId,
      'recordName': recordName,
      'reportDate': reportDate.toUtc().toIso8601String(),
      if (comments != null && comments!.trim().isNotEmpty)
        'comments': comments!.trim(),
    };
  }
}

File: backend/src/Contracts/Patients/UpdatePatientMedicalRecordRequest.cs
namespace OptimaHealthcare.Contracts.Patients;

public sealed class UpdatePatientMedicalRecordRequest
{
    public int RecordTypeId { get; set; }
    public string RecordName { get; set; } = string.Empty;
    public DateTime ReportDate { get; set; }
    public string? Comments { get; set; }
}
